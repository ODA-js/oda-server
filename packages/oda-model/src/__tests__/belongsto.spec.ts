import 'jest';
import { BelongsTo } from '../';

describe('belongsto ', () => {
  it('loads', () => {
    const bt = new BelongsTo({
      belongsTo: 'A#',
      entity: 'B',
      field: 'a',
    });
  });
});
