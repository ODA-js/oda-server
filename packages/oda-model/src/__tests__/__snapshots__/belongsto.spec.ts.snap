// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`belongsto  loads 1`] = `
BelongsTo {
  "$obj": Object {
    "belongsTo": EntityReference {
      "$obj": Object {
        "backField": "",
        "entity": "A",
        "field": "id",
      },
      "metadata_": Object {},
    },
    "entity": "B",
    "field": "a",
  },
  "metadata_": Object {
    "name": Object {
      "full": "BBelongsToA",
      "normal": "BA",
      "short": "A",
    },
    "persistence": Object {
      "embedded": false,
      "identity": false,
      "indexed": true,
      "required": false,
      "single": true,
      "stored": true,
    },
    "verb": "BelongsTo",
  },
}
`;

exports[`belongsto  loads 2`] = `
Object {
  "belongsTo": "A#id",
  "embedded": false,
  "entity": "B",
  "field": "a",
  "metadata": Object {
    "name": Object {
      "full": "BBelongsToA",
      "normal": "BA",
      "short": "A",
    },
    "persistence": Object {
      "embedded": false,
      "identity": false,
      "indexed": true,
      "required": false,
      "single": true,
      "stored": true,
    },
    "verb": "BelongsTo",
  },
  "name": undefined,
  "opposite": undefined,
}
`;
